### PR TITLE
Simplify configuration properties

### PR DESCRIPTION
**Pull request by gboyvalenkov-bosch**
[#20] Simplify configuration properties

- code refactoring and optimizations

Signed-off-by: Georgi Boyvalenkov <Georgi.Boyvalenkov@bosch.io>

### CODE REVIEW

File: flagparse/flags.go [hunk 1]
```diff
@@ -32,49 +32,36 @@ import (
...
+	logger.LogConfig
 
-	Files string `json:"files,omitempty"`
+	Files string `json:"files,omitempty" descr:"Glob pattern for the files to upload"`
+}
+
+//ConfigNames contains template names to be replaced in config properties descriptions and default values
+var ConfigNames = map[string]string{
+	"name": "Autouploadable", "feature": "Uploadable", "period": "Upload period",
+	"action": "upload", "actions": "uploads", "running_actions": "uploads",
```
Comment (e-grigorov) [inline flagparse/flags.go:50]: Do we need actions and running_actions when the values are the same?

File: flagparse/flags.go [hunk 2]
```diff
@@ -32,49 +32,36 @@ import (
...
 }
 
 //ConfigFileMissing error, which represents a warning for missing config file
 type ConfigFileMissing error
 
-//NewUploadFileConfig return initialized UploadFileConfig
```
Comment (e-grigorov) [inline flagparse/flags.go:51]: typo: return should be returns

File: flagparse/flags.go [hunk 3]
```diff
@@ -98,44 +86,42 @@ func ParseFlags(version string) (*client.BrokerConfig, *client.UploadableConfig,
 		}
 	}
 
-	return brokerConfig, uploadConfig, logConfig, filesGlob, warn
+	return config, warn
 }
 
 // ApplyFlags applies cli values over config values
-func ApplyFlags(config *UploadFileConfig, brokerConfig *client.BrokerConfig,
```
Comment (e-grigorov) [inline flagparse/flags.go:105]: There is no real need to have it as a public method.

File: flagparse/flags.go [hunk 4]
```diff
@@ -98,44 +86,42 @@ func ParseFlags(version string) (*client.BrokerConfig, *client.UploadableConfig,
...
 	})
 }
 
-func applyConfigurationFile(configFile string, brokerConfig *client.BrokerConfig,
-	uploadConfig *client.UploadableConfig, logConfig *logger.LogConfig, filesGlob *string) ConfigFileMissing {
-	def := GetUploadFileConfigDefaults()
+//InitFlagVars parses the 'cfg' structure and defines flag variables for its fields.
+//The 'cfg' parameter should be a pointer to structure. Flag names are taken from field names (with the first letter lower cased).
+//Flag descriptions are taken from 'descr' field tags, default values are taken from 'def' field tags
+func InitFlagVars(cfg interface{}, names map[string]string, skip map[string]bool) {
```
Comment (e-grigorov) [inline flagparse/flags.go:111]: Do we need it as a public method?

File: flagparse/flags.go [hunk 5]
```diff
@@ -171,7 +172,7 @@ func initConfigValues(typeOfConfig reflect.Type, valueOfConfig reflect.Value, fl
 		case bool:
 			defaultBoolValue, err := strconv.ParseBool(defaultValue)
 			if err != nil {
-				log.Println(fmt.Sprintf("Error parsing boolean argument %v with value %v", fieldType.Name, defaultValue))
+				log.Printf("Error parsing boolean argument %v with value %v", fieldType.Name, defaultValue)
```
Comment (e-grigorov) [inline flagparse/flags.go:175]: A few points here:
* Is it possible at all? The default values should be valid.
* The message should say that it's the default value.
* Is it enough to have only log?

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: flagparse/flags.go (hunk 5 above)
Comment (gboyvalenkov-bosch) [reply flagparse/flags.go:175]: This code handles incorrect definition(s) of flag default values. Maybe it should return an error/panic?

File: flagparse/flags.go (hunk 1 above)
Comment (gboyvalenkov-bosch) [reply flagparse/flags.go:50]: Yes, it will be used for backup, where actions are different.

File: flagparse/flags.go [hunk 6]
```diff
@@ -32,49 +32,36 @@ import (
 const (
 	ConfigFile = "configFile"
 
-	Files        = "files"
-	DefaultFiles = ""
+	Files = "files"
 )
 
 //UploadFileConfig describes file config of uploadable
 type UploadFileConfig struct {
```
Comment (e-grigorov) [inline flagparse/flags.go:39]: This struct is not "file" config only. It's a generic upload configuration description. What about: UploadConfig?

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: flagparse/flags.go (hunk 5 above)
Comment (gboyvalenkov-bosch) [reply flagparse/flags.go:179]: Separate issue has been posted - [#25].

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**